### PR TITLE
[CAP 35] Add support for clawback account and trustline flags

### PR DESCRIPTION
**Pull request by 2opremio**
Add support for the new flags in CAP 35

Part of #3348 

Depends on #3350

### CODE REVIEW

File: services/horizon/internal/ingest/processors/operations_processor.go [hunk 1]
```diff
@@ -351,8 +351,13 @@ func (operation *transactionOperationWrapper) Details() (map[string]interface{},
...
 		details["authorize"] = xdr.TrustLineFlags(op.Authorize).IsAuthorized()
-		if xdr.TrustLineFlags(op.Authorize).IsAuthorizedToMaintainLiabilitiesFlag() {
-			details["authorize_to_maintain_liabilities"] = xdr.TrustLineFlags(op.Authorize).IsAuthorizedToMaintainLiabilitiesFlag()
+		authLiabilities := xdr.TrustLineFlags(op.Authorize).IsAuthorizedToMaintainLiabilitiesFlag()
+		if authLiabilities {
+			details["authorize_to_maintain_liabilities"] = authLiabilities
+		}
+		clawbackEnabled := xdr.TrustLineFlags(op.Authorize).IsClawbackEnabledFlag()
+		if clawbackEnabled {
+			details["claw_back_enabled"] = clawbackEnabled
```
Comment (bartekn) [inline services/horizon/internal/ingest/processors/operations_processor.go:360]: It's _clawback_ everywhere else.
```suggestion
			details["clawback_enabled"] = clawbackEnabled
```

File: txnbuild/operation_test.go [hunk 2]
```diff
@@ -214,8 +214,8 @@ func TestSetOptionsFromXDR(t *testing.T) {
 	var opSource xdr.AccountId
 	err := opSource.SetAddress("GB7BDSZU2Y27LYNLALKKALB52WS2IZWYBDGY6EQBLEED3TJOCVMZRH7H")
 	assert.NoError(t, err)
-	cFlags := xdr.Uint32(5)
-	sFlags := xdr.Uint32(7)
+	cFlags := xdr.Uint32(0xD)
+	sFlags := xdr.Uint32(0xF)
```
Comment (bartekn) [inline txnbuild/operation_test.go:218]: Nit, for readability.
```suggestion
	cFlags := xdr.Uint32(0b1101)
	sFlags := xdr.Uint32(0b1111)
```

File: xdr/Stellar-transaction.x [hunk 3]
```diff
@@ -867,7 +886,8 @@ enum SetOptionsResultCode
     SET_OPTIONS_UNKNOWN_FLAG = -6,           // can't set an unknown flag
     SET_OPTIONS_THRESHOLD_OUT_OF_RANGE = -7, // bad value for weight/threshold
     SET_OPTIONS_BAD_SIGNER = -8,             // signer cannot be masterkey
-    SET_OPTIONS_INVALID_HOME_DOMAIN = -9     // malformed home domain
+    SET_OPTIONS_INVALID_HOME_DOMAIN = -9,     // malformed home domain
+    SET_OPTIONS_AUTH_REVOCABLE_REQUIRED = -10 // auth revocable is required for clawback
```
Comment (bartekn) [inline xdr/Stellar-transaction.x:890]: We may need to update error codes returned by txsub. Worth adding an integration test.

**Review by bartekn — APPROVED** [review]

File: services/horizon/internal/ingest/processors/operations_processor.go (hunk 1 above)
Comment (2opremio) [reply services/horizon/internal/ingest/processors/operations_processor.go:360]: Good catch

File: xdr/Stellar-transaction.x (hunk 3 above)
Comment (2opremio) [reply xdr/Stellar-transaction.x:890]: I will add it to #3348

**Pull request merged**